universal-application-tool-0.0.1/test/controllers/admin/QuestionControllerTest.java

```diff
@@ -0,0 +1,159 @@
+    controller
+        .edit(request, "the.ultimate.question")
+        .thenAccept(
+            result -> {
+              assertThat(result.status()).isEqualTo(OK);
+              assertThat(contentAsString(result)).contains("Edit Question");
+            });
+  }
+
+  @Test
+  public void index_returnsQuestions() throws UnsupportedQuestionTypeException {
+    buildQuestionsList();
+    controller
+        .index("table")
+        .thenAccept(
+            result -> {
+              assertThat(result.status()).isEqualTo(OK);
+              assertThat(result.contentType()).hasValue("text/html");
+              assertThat(result.charset()).hasValue("utf-8");
+              assertThat(contentAsString(result)).contains("Total Questions: 1");
+              assertThat(contentAsString(result)).contains("All Questions");
+            });
+  }
+
+  @Test
+  public void index_withNoQuestions() {
+    controller
+        .index("table")
+        .thenAccept(
+            result -> {
+              assertThat(result.status()).isEqualTo(OK);
+              assertThat(result.contentType()).hasValue("text/html");
+              assertThat(result.charset()).hasValue("utf-8");
+              assertThat(contentAsString(result)).contains("Total Questions: 0");
+              assertThat(contentAsString(result)).contains("All Questions");
+            });
+  }
+
+  @Test
+  public void newOne_returnsExpectedForm() {
+    Request request = addCSRFToken(Helpers.fakeRequest()).build();
+    Result result = controller.newOne(request);
+
+    assertThat(result.status()).isEqualTo(OK);
+    assertThat(contentAsString(result)).contains("New Question");
+    assertThat(contentAsString(result)).contains(CSRF.getToken(request.asScala()).value());
+  }
+
+  @Test
+  public void update_failsGracefully() throws UnsupportedQuestionTypeException {
+    // TODO: Update isn't implemented yet, so we just redirect to the questions list for now.
+    // https://github.com/seattle-uat/universal-application-tool/issues/103
+    buildQuestionsList();
+    ImmutableMap.Builder<String, String> formData = ImmutableMap.builder();
+    formData
+        .put("questionName", "name")
+        .put("questionDescription", "desc")
+        .put("questionPath", "my.question.path")
+        .put("questionType", "TEXT")
+        .put("questionText", "Hi mom!")
+        .put("questionHelpText", ":-)");
+    RequestBuilder requestBuilder = Helpers.fakeRequest().bodyForm(formData.build());
+    controller
+        .update(requestBuilder.build(), 1L)
+        .thenAccept(
+            result -> {
+              assertThat(contentAsString(result)).contains("Total Questions: 1");
+              assertThat(contentAsString(result)).contains("All Questions");
+            });
+  }
+
+  private void buildQuestionsList() throws UnsupportedQuestionTypeException {
+    QuestionDefinitionBuilder builder =
+        new QuestionDefinitionBuilder()
+            .setId(1L)
```

Comment:
nit (blocking): we can skip setting id in builder now. The default value is empty.

Reply:
Done.